src/modules/gatsby-plugin/createImgixGatsbyImageDataFieldConfig.ts

```diff
@@ -59,94 +66,62 @@ const resolveGatsbyImageData = <TSource>({
-        buildGatsbyImageDataBaseArgs({
-          url,
-          dimensions,
-          resolverArgs: safeResolverArgs,
-          defaultParams,
-          imgixClient,
-        }),
-      )
-      .bindL(
-        'placeholderData',
-        ({ url, baseImageDataArgs, safeResolverArgs }) => {
-          if (safeResolverArgs.placeholder === 'blurred') {
-            return pipe(
-              getLowResolutionImageURL({
-                ...baseImageDataArgs,
-                options: {
-                  ...baseImageDataArgs,
-                  imgixParams: {
-                    ...defaultParams,
-                    ...safeResolverArgs.imgixParams,
-                    ...safeResolverArgs.placeholderImgixParams,
-                  },
-                },
-              }),
-              fetchImgixBase64Image(cache),
-              TE.map((base64Data) => ({
-                placeholder: { fallback: base64Data },
-              })),
-            );
-          }
-          if (safeResolverArgs.placeholder === 'dominantColor') {
-            return pipe(
-              fetchImgixDominantColor(cache)((params) =>
-                imgixClient.buildURL(url, {
-                  ...defaultParams,
-                  ...safeResolverArgs.imgixParams,
-                  ...safeResolverArgs.placeholderImgixParams,
-                  ...params,
-                }),
-              ),
-              TE.map((dominantColor) => ({
-                backgroundColor: dominantColor,
-              })),
-            );
-          }
-          return TE.right({});
-        },
-      )
-      .return(({ baseImageDataArgs, placeholderData }) => ({
-        ...generateImageData({
-          ...baseImageDataArgs,
-        }),
-        ...placeholderData,
-      })),
-    TE.getOrElseW(() => T.of(undefined)),
-  )();
+  try {
+    const url = await resolveUrl(rootValue);
+    if (!url) {
+      return undefined;
+    }
+    const manualWidth = await resolveWidth(rootValue);
+    const manualHeight = await resolveHeight(rootValue);
+
+    const safeResolverArgs = {
+      ...unsafeResolverArgs,
+      imgixParams: unTransformParams(unsafeResolverArgs.imgixParams ?? {}),
+      placeholderImgixParams: unTransformParams(
+        unsafeResolverArgs.placeholderImgixParams ?? {},
+      ),
+    };
+
+    const dimensions = await TE.getOrElseW<Error, IResolveDimensionsRight>(
+      () => {
+        throw new Error('Something went wrong while resolving dimensions');
+      },
+    )(
+      resolveDimensions({
+        url,
+        manualHeight: O.fromNullable(manualHeight),
+        manualWidth: O.fromNullable(manualWidth),
+        cache,
+        client: imgixClient,
+      }),
+    )();
+
+    const baseImageDataArgs = buildGatsbyImageDataBaseArgs({
+      url,
+      dimensions,
+      resolverArgs: safeResolverArgs,
+      defaultParams,
+      imgixClient,
+    });
+
+    const placeholderData = await generatePlaceHolderData({
```

Comment:
nit: maybe a comment here or after 251 explaining why we need the placeholder data would be useful. As much as I _think_ I follow, probably not grasping this to its fullest.

Reply:
👍